### PR TITLE
Integrating authn into gateway

### PR DESCRIPTION
**Pull request by HJianBo**

### CODE REVIEW

File: apps/emqx_gateway/src/emqx_gateway_schema.erl [hunk 1]
```diff
@@ -195,42 +197,41 @@ fields(dtls_opts) ->
...
-%       , hoconsc:ref(emqx_authn_http, get)
-%       , hoconsc:ref(emqx_authn_http, post)
-%       , hoconsc:ref(emqx_authn_jwt, 'hmac-based')
-%       , hoconsc:ref(emqx_authn_jwt, 'public-key')
-%       , hoconsc:ref(emqx_authn_jwt, 'jwks')
-%       , hoconsc:ref(emqx_enhanced_authn_scram_mnesia, config)
-%       ]).
+authentication() ->
+    hoconsc:union(
+      [ undefined
```
Comment (zmstone) [inline apps/emqx_gateway/src/emqx_gateway_schema.erl:202]: make the field nullable instead of union with `undefined`

Comment (zmstone) [reply apps/emqx_gateway/src/emqx_gateway_schema.erl:202]: `hoconsc:mk(hoconsc:union([....], #{nullable => true, desc => "..."}`)

File: apps/emqx_gateway/src/emqx_gateway_insta_sup.erl [hunk 2]
```diff
@@ -197,52 +205,102 @@ detailed_gateway_info(State) ->
...
-                    %            {ok, _} -> ChainId;
-                    %            {error, Reason} ->
-                    %                ?LOG(error, "Failed to create authentication ~p", [Reason]),
-                    %                throw({bad_authentication, Reason})
-                    %        end;
-                    %    {error, Reason} ->
-                    %        ?LOG(error, "Failed to create authentication chain: ~p", [Reason]),
-                    %        throw({bad_chain, {ChainId, Reason}})
-                    %end.
+%% same with emqx_authentication:global_chain/1
```
Comment (tigercl) [inline apps/emqx_gateway/src/emqx_gateway_insta_sup.erl:208]: export emqx_authentication:global_chain/1 please and use it in here

**Review by tigercl — APPROVED** [review]

File: apps/emqx_gateway/src/emqx_gateway_insta_sup.erl (hunk 2 above)
Comment (HJianBo) [reply apps/emqx_gateway/src/emqx_gateway_insta_sup.erl:208]: I'll improve it in the later PR

**Pull request merged**